rnote-ui/src/workspacebrowser/filerow/actions/rename.rs

```diff
@@ -0,0 +1,103 @@
+use std::path::PathBuf;
+
+use gtk4::{
+    gio, glib,
+    glib::clone,
+    pango,
+    prelude::FileExt,
+    traits::{ButtonExt, EditableExt, PopoverExt, WidgetExt, StyleContextExt},
+    Align, Button, Entry, Popover, Label,
+};
+
+use gettextrs::gettext;
+
+use crate::workspacebrowser::{FileRow, filerow::widget_helper};
+
+impl FileRow {
+    pub fn rename_action(&self) -> gio::SimpleAction {
+        let rename_action = gio::SimpleAction::new("rename-file", None);
+
+        rename_action.connect_activate(clone!(@weak self as filerow => move |_action_rename_file, _| {
+            if let Some(current_file) = filerow.current_file() {
+                if let Some(current_path) = current_file.path() {
+                    if let Some(parent_path) = current_path.parent().map(|parent_path| parent_path.to_path_buf()) {
+
+                        let entry = get_entry(&current_path);
+                        let label = get_label();
+
+                        let (grid, cancel_button, apply_button, popover) = widget_helper::entry_dialog::get_entry_dialog(&entry, &label);
+
+                        connect_entry(&entry, &apply_button, parent_path.clone());
+                        connect_apply_button(&apply_button, &popover, &entry, parent_path.clone(),
+                            current_file.clone());
+
+                        log::debug!("Start popup");
+                        popover.popup();
```

Comment:
@flxzt I don't know if this is possible but I might need your help here (if possible).

Currently `rnote` is crashing exactly at this line and I don't understand why. Those are the critical error messages:
```
(rnote:3171): Gtk-CRITICAL **: 02:44:57.395: gtk_widget_get_native: assertion 'GTK_IS_WIDGET (widget)' failed

(rnote:3171): Gtk-CRITICAL **: 02:44:57.395: gtk_native_get_surface: assertion 'GTK_IS_NATIVE (self)' failed

(rnote:3171): Gdk-CRITICAL **: 02:44:57.395: gdk_surface_new_popup: assertion 'GDK_IS_SURFACE (parent)' failed
[1]    3171 segmentation fault  RUST_LOG=debug rnote
```

I basically moved the rename-code-block into different functions. I assume, since the error messages says something isn't a widget, that a widget of `popover` like the grid or something like that gets freed before but I might be wrong, do you have any ideas?

Reply:
Hm... ok I got the renaming working again by pasting your old code into my function back again... I'll keep investigating this.

Reply:
Ok, the compiler is probably doing some nasty thinks which aren't working out in the binary ;-;

Reply:
I think you have to add the popover to the widget tree somewhere: https://github.com/flxzt/rnote/pull/276/files/6ebb828832c74a933598bbf5d736d7098bb57470#diff-048da3f13be4da1d8ea47c267534bf64d2082f7c315b1cf6205f15307f532146L249

Reply:
Yesss!!!!! That was the issue!! :) Thanks!